test/controllers/payment.controller.spec.ts

```diff
@@ -41,63 +46,88 @@ mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, ne
 
-  test("should rejecting redirect, state does not match", async () => {
-    mockGetApplicationData.mockReturnValueOnce( {} );
-    await request(app).get(PAYMENT_WITH_TRANSACTION_URL_AND_QUERY_STRING);
-
-    expect(mockLoggerInfoRequest).toHaveBeenCalledTimes(1);
-    expect(mockLoggerDebugRequest).not.toHaveBeenCalled();
-    expect(mockCreateAndLogErrorRequest).toHaveBeenCalledTimes(1);
-  });
-
-  test(`should redirect to ${CONFIRMATION_PAGE} page, Payment Successful with status ${PAYMENT_PAID}`, async () => {
-    mockGetApplicationData.mockReturnValueOnce( { [PaymentKey]: PAYMENT_OBJECT_MOCK } );
-    const resp = await request(app).get(PAYMENT_WITH_TRANSACTION_URL_AND_QUERY_STRING);
-
-    expect(resp.status).toEqual(302);
-    expect(resp.text).toEqual(`${FOUND_REDIRECT_TO} ${CONFIRMATION_URL}`);
-    expect(mockLoggerDebugRequest).toHaveBeenCalledTimes(1);
-    expect(mockLoggerInfoRequest).toHaveBeenCalledTimes(1);
-    expect(mockCreateAndLogErrorRequest).not.toHaveBeenCalled();
-  });
-
-  test(`should redirect to ${CHECK_YOUR_ANSWERS_URL} page, Payment failed somehow`, async () => {
-    mockGetApplicationData.mockReturnValueOnce( { [PaymentKey]: PAYMENT_OBJECT_MOCK } );
-    const resp = await request(app).get(PAYMENT_DECLINED_WITH_TRANSACTION_URL_AND_QUERY_STRING);
-
-    expect(resp.status).toEqual(302);
-    expect(resp.text).toEqual(`${FOUND_REDIRECT_TO} ${CHECK_YOUR_ANSWERS_URL}`);
-    expect(mockLoggerDebugRequest).toHaveBeenCalledTimes(1);
-    expect(mockLoggerInfoRequest).toHaveBeenCalledTimes(1);
-    expect(mockCreateAndLogErrorRequest).not.toHaveBeenCalled();
-  });
-
-  test(`should redirect to ${PAYMENT_FAILED_PAGE} page, Payment failed somehow and feature flag active`, async () => {
-    mockIsActiveFeature.mockReturnValueOnce(true);
-    mockGetApplicationData.mockReturnValueOnce( { [PaymentKey]: PAYMENT_OBJECT_MOCK } );
-    const resp = await request(app).get(PAYMENT_DECLINED_WITH_TRANSACTION_URL_AND_QUERY_STRING);
-
-    expect(resp.status).toEqual(302);
-    expect(resp.text).toEqual(`${FOUND_REDIRECT_TO} ${PAYMENT_FAILED_URL}`);
-    expect(mockLoggerDebugRequest).toHaveBeenCalledTimes(1);
-    expect(mockLoggerInfoRequest).toHaveBeenCalledTimes(1);
-    expect(mockCreateAndLogErrorRequest).not.toHaveBeenCalled();
+  describe("GET tests for Payment controller without params url", () => {
+    test("should rejecting redirect, state does not match", async () => {
+      mockGetApplicationData.mockReturnValueOnce( {} );
+      await request(app).get(PAYMENT_WITH_TRANSACTION_URL_AND_QUERY_STRING);
+
+      expect(mockLoggerInfoRequest).toHaveBeenCalledTimes(1);
+      expect(mockLoggerDebugRequest).not.toHaveBeenCalled();
+      expect(mockCreateAndLogErrorRequest).toHaveBeenCalledTimes(1);
+    });
+
+    test(`should redirect to ${CONFIRMATION_PAGE} page, Payment Successful with status ${PAYMENT_PAID}`, async () => {
+      mockGetApplicationData.mockReturnValueOnce( { [PaymentKey]: PAYMENT_OBJECT_MOCK } );
```

Comment:
We didn't copy most of these tests because they weren't required? The lines appear as new changes because we indented when adding a new code block below them, which is why Git is displaying them as changes.